src/sinks/aws_kinesis_firehose.rs

```diff
@@ -50,23 +52,22 @@ lazy_static! {
     };
 }
 
-#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone, Derivative)]
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone)]
 #[serde(rename_all = "snake_case")]
-#[derivative(Default)]
 pub enum Encoding {
-    #[derivative(Default)]
     Text,
     Json,
 }
 
 inventory::submit! {
-    SinkDescription::new::<KinesisFirehoseSinkConfig>("aws_kinesis_firehose")
+    SinkDescription::new_without_default::<KinesisFirehoseSinkConfig>("aws_kinesis_firehose")
 }
 
 #[typetag::serde(name = "aws_kinesis_firehose")]
 impl SinkConfig for KinesisFirehoseSinkConfig {
     fn build(&self, cx: SinkContext) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
         let config = self.clone();
+        config.encoding.validate()?;
```

Comment:
since it looks like we call this on each build we could just nest this into our de impl for EncodingConfig?